### PR TITLE
[trivial] Added immutable to, and got rid of several variables in std.encoding

### PR DESCRIPTION
**Pull request by JackStouffer**

### CODE REVIEW

File: std/encoding.d [hunk 1]
```diff
@@ -1660,8 +1659,7 @@ immutable(E)[] sanitize(E)(immutable(E)[] s)
     t = s[n..$];
     while (t.length != 0)
     {
-        dchar c = EncoderInstance!(E).safeDecode(t);
-        assert(c == INVALID_SEQUENCE);
+        assert(EncoderInstance!(E).safeDecode(t) == INVALID_SEQUENCE);
```
Comment (schveiguy) [inline std/encoding.d]: No, because this will get compiled out in release mode! you have to keep the verbose version.

**Comment by JackStouffer** [comment]
@schveiguy Fixed

**Comment by schveiguy** [comment]
why the inconsistency between `const` and `immutable`?

**Comment by JackStouffer** [comment]
Mistake. I was thinking in terms of my other PR.

@schveiguy Fixed

**Comment by schveiguy** [comment]
LGTM. Awaiting Travis...

**Comment by WalterBright** [comment]
Auto-merge toggled on

**Pull request merged**